tests/system/test_diags.py

```diff
@@ -331,8 +334,9 @@ def test_results_dir(self):
 
     # Test the image count
     def test_image_count(self):
-        actual_num_images = count_images("all_sets_results_test")
-        expected_num_images = count_images("../unit_test_images")
+        actual_num_images, actual_images = count_images("all_sets_results_test")
+        expected_num_images, expected_images = count_images("../unit_test_images")
+        self.assertEqual(actual_images, expected_images)
```

Comment:
Added an assertion that the image names match -- this helps us determine which images are extra and/or missing.